CMakeLists.txt

```diff
@@ -46,7 +46,7 @@ option(POLYHOOK_FEATURE_VIRTUALS "Implement all virtual table hooking functional
 
 function(add_asmjit_properties)
     if(MSVC)
-        if(ASMJIT_STATIC)
+        if(POLYHOOK_BUILD_STATIC_RUNTIME)
```

Comment:
I don't believe this is the right flag. The polyhook runtime flag is meant for polyhook itself, we have separate flags for the linked libraries:

* `POLYHOOK_BUILD_SHARED_ASMTK`
* `POLYHOOK_BUILD_SHARED_ASMJIT`
* `POLYHOOK_BUILD_SHARED_ZYDIS`

The ASMJIT_STATIC flag is from the docs here for reference: https://asmjit.com/doc/group__asmjit__build.html#gabcd1a6399d893d32c3bc5c8419c7a694

Reply:
I think `POLYHOOK_BUILD_SHARED_ASMJIT` would be the flag we actually want. As is I believe this would work out correctly by accident because when a user sets ASMJIT_STATIC to TRUE they should always have `POLYHOOK_BUILD_SHARED_ASMJIT` set to false anyways.

Reply:
The problem though is not if AsmJit is statically linked or not, but rather the runtime libraries themselves. Without this change, building PolyHook with the dynamic runtime libraries causes AsmJit to still expect statically linked runtime libraries when AsmJit itself is linked statically.

Reply:
You should set POLYHOOK_USE_EXTERNAL_ASMJIT  and POLYHOOK_BUILD_SHARED_ASMJIT separately from how the library is linked. It's a good default configuration to have polyhook generate a single dynamic dll that statically links all of it's dependencies

Reply:
No, I was trying to statically link PolyHook and all of its dependencies, with the exception of the runtime libraries. The runtime libraries (and only the runtime libraries) need to be dynamically linked for my purposes.

Reply:
I understand the issue now. I think you're correct. We'll need to be careful this works with the vcpkg port